server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -450,6 +437,9 @@ synchronized void updateRemoteCluster(
             connectionMap.put(clusterAlias, remote);
             remote.ensureConnected(listener.map(ignored -> RemoteClusterConnectionStatus.RECONNECTED));
         } else {
+            if (crossProjectEnabled == false) {
+                remote.setSkipUnavailable(config.skipUnavailable());
```

Comment:
To be strictly faithful to existing behaviour, I think this should be something like
```suggestion
                if (remote.isSkipUnavailable() != config.skipUnavailable()) {
                    remote.setSkipUnavailable(config.skipUnavailable());
                }
```